contracts/Relay.sol

```diff
@@ -283,19 +284,27 @@ contract Relay is Ownable, Pausable, Initializable {
         bytes memory message,
         bytes[] memory signatures
     ) internal view returns (bool) {
+        require(signatures.length != 0, "Relay:: _checkSignature: signatures is empty");
         bytes32 hash = keccak256(message);
-        require(signatures.length < 0xffffffff, "Relay: overflow");
+        uint256 count;
+        address[] memory signers = new address[](signatures.length);
+        bytes32 ethSignedMessageHash = ECDSA.toEthSignedMessageHash(hash);
 
-        uint16 count;
-        for (uint16 i = 0; i < signatures.length; i++) {
-            address signer = ECDSA.recover(ECDSA.toEthSignedMessageHash(hash), signatures[i]);
-            if (isRelayer(signer)) {
-                count++;
-            }
+        for (uint256 i = 0; i < signatures.length; i++) {
+            address signer = ECDSA.recover(ethSignedMessageHash, signatures[i]);
+            signers[i] = signer;
         }
 
+        checkDuplicates(signers);
```

Comment:
solidity does not support mapping memory type, right?

Reply:
mapping type only supports storage